src/Symfony/Component/DependencyInjection/Tests/Loader/FileLoaderTest.php

```diff
@@ -274,6 +282,73 @@ public function testRegisterClassesWithWhenEnv(?string $env, bool $expected)
 
         $this->assertSame($expected, $container->has(Foo::class));
     }
+
+    /**
+     * @dataProvider provideResourcesWithAsAliasAttributes
+     */
+    public function testRegisterClassesWithAsAlias(string $resource, array $expectedAliases)
+    {
+        $container = new ContainerBuilder();
+        $loader = new TestFileLoader($container, new FileLocator(self::$fixturesPath.'/Fixtures'));
+        $loader->registerClasses(
+            (new Definition())->setAutoconfigured(true),
+            'Symfony\Component\DependencyInjection\Tests\Fixtures\PrototypeAsAlias\\',
+            $resource
+        );
+
+        $this->assertEquals($expectedAliases, $container->getAliases());
+    }
+
+    public static function provideResourcesWithAsAliasAttributes(): iterable
+    {
+        yield 'Private' => ['PrototypeAsAlias/{WithAsAlias,AliasFooInterface}.php', [AliasFooInterface::class => new Alias(WithAsAlias::class)]];
+        yield 'Interface' => ['PrototypeAsAlias/{WithAsAliasInterface,AliasFooInterface}.php', [AliasFooInterface::class => new Alias(WithAsAliasInterface::class)]];
+        yield 'Multiple' => ['PrototypeAsAlias/{WithAsAliasMultiple,AliasFooInterface}.php', [
+            AliasFooInterface::class => new Alias(WithAsAliasMultiple::class, true),
+            'some-alias' => new Alias(WithAsAliasMultiple::class),
+        ]];
+        yield 'Multiple with id' => ['PrototypeAsAlias/{WithAsAliasIdMultipleInterface,AliasBarInterface,AliasFooInterface}.php', [
+            AliasBarInterface::class => new Alias(WithAsAliasIdMultipleInterface::class),
+            AliasFooInterface::class => new Alias(WithAsAliasIdMultipleInterface::class),
```

Comment:
Why is the `AliasFooInterface` being registered? Is it because of the `WithAsAliasIdMultipleInterface` class being the only one implementing this interface?

Reply:
Yes. I think it's the right behavior, don't you think so? The other possibility would be to prevent the singly implemented interface registration if there is an `AsAlias` attribute on the class.